core/src/main/java/com/avast/bytes/Bytes.java

```diff
@@ -202,17 +202,17 @@ static Bytes copyFromHex(String hexString) {
         if (length % 2 != 0) {
             throw new IllegalArgumentException("HexString needs to be even-length: " + hexString);
         }
-        byte[] result = new byte[length / 2];
-        for (int i = 0; i < length; i += 2) {
-            int high = Character.getNumericValue(hexString.charAt(i));
-            int low = Character.getNumericValue(hexString.charAt(i + 1));
-            if (high == -1 || low == -1) {
-                throw new IllegalArgumentException("HexString contains illegal characters: " + hexString);
+
+        try {
+            byte[] result = new byte[length / 2];
+            for (int i = 0; i < length; i += 2) {
+                int high = Utils.hexToBin(hexString.charAt(i));
+                int low = Utils.hexToBin(hexString.charAt(i + 1));
+                result[i >> 1] = (byte) ((high << 4) + low);
             }
-            result[i / 2] = (byte) (high * 16 + low);
+            return copyFrom(result);
+        } catch (IllegalArgumentException e) {
```

Comment:
Isn't this kind of useless?

Reply:
It adds context to the exception which may significantly help during debugging (logs), but I don't insist on it.

Reply:
I think it just  pollutes the stacktrace. But I don't care that much so it's OK from my POV.

Reply:
@mixalturek 👍 i like it, keep it in